CrossSectionAnalysis/CrossSectionAnalysis.py

```diff
@@ -0,0 +1,422 @@
+import os
+import unittest
+import logging
+import vtk, qt, ctk, slicer
+from slicer.ScriptedLoadableModule import *
+import numpy
+from slicer.util import VTKObservationMixin
+
+#
+# CrossSectionAnalysis
+# renamed from
+# PathReformat
+#
+
+class CrossSectionAnalysis(ScriptedLoadableModule):
+  """Uses ScriptedLoadableModule base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent):
+    ScriptedLoadableModule.__init__(self, parent)
+    self.parent.title = "Cross-section analysis"  # TODO: make this more human readable by adding spaces
+    self.parent.categories = ["Vascular Modeling Toolkit"]  # TODO: set categories (folders where the module shows up in the module selector)
+    self.parent.dependencies = []  # TODO: add here list of module names that this module requires
+    self.parent.contributors = ["SET (Hobbyist)"]  # TODO: replace with "Firstname Lastname (Organization)"
+    # TODO: update with short description of the module and a link to online module documentation
+    self.parent.helpText = """
+Moves a selected view along a path, and orients the plane at right angle to the path. It is intended to view cross-sections of blood vessels.
+See more information in <a href="https://github.com/chir-set/PathReformat">module documentation</a>.
+"""
+    # TODO: replace with organization, grant and thanks
+    self.parent.acknowledgementText = """
+This file was originally developed by Jean-Christophe Fillion-Robin, Kitware Inc., Andras Lasso, PerkLab,
+and Steve Pieper, Isomics, Inc. and was partially funded by NIH grant 3P41RR013218-12S1.
+"""
+
+#
+# CrossSectionAnalysisWidget
+#
+
+class CrossSectionAnalysisWidget(ScriptedLoadableModuleWidget, VTKObservationMixin):
+  """Uses ScriptedLoadableModuleWidget base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent=None):
+    """
+    Called when the user opens the module the first time and the widget is initialized.
+    """
+    ScriptedLoadableModuleWidget.__init__(self, parent)
+    VTKObservationMixin.__init__(self)  # needed for parameter node observation
+    self.logic = None
+    # Widget level observers to update module UI. Logic class has its own.
+    self.widgetMarkupPointObserver = None
```

Comment:
> you never _distinguish_ between them

Please clarify.
Three variables are **distinctly** declared though : 
self.widgetMarkupPointObserver, self.widgetMarkupPointAddedObserver and self.widgetMarkupPointRemovedObserver.

Reply:
There is no need to distinguish between the observations. You can store all 3 in a list.  A list is just a single variable instead of 3, and it is easier to manage the (e.g., you can simply iterate through them instead of naming them one by one).

Reply:
Curve point observations are now packed in lists. Not sure if that's what you meant though.

Reply:
Yes, it's good now.